src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java

```diff
@@ -62,7 +90,23 @@ public CassandraOutgoingFile(StreamOperation operation, Ref<SSTableReader> ref,
                                                 sections,
                                                 sstable.compression ? sstable.getCompressionMetadata() : null,
                                                 keepSSTableLevel ? sstable.getSSTableLevel() : 0,
-                                                sstable.header.toComponent());
+                                                sstable.header.toComponent(), manifest, shouldStreamFullSSTable(),
+                                                sstable.first,
+                                                sstable.metadata().id);
+    }
+
+    @VisibleForTesting
+    public static ComponentManifest getComponentManifest(SSTableReader sstable)
+    {
+        LinkedHashMap<Component.Type, Long> components = new LinkedHashMap<>(STREAM_COMPONENTS.size());
+        for (Component component : STREAM_COMPONENTS)
```

Comment:
@iamaleksey would it be worth having a list of required components here? Any one of the missing components will be a fatal? For example, when I reset the level on the receiving side, I expect stats to exist and is sent over.